project/project.go

```diff
@@ -129,12 +129,16 @@ func (p *State) Schema(ctx context.Context, gw bkgw.Client, platform specs.Platf
 			return
 		}
 
+		session = session.WithLocalDirs(fsPayload.LocalDirs)
```

Comment:
This is one trade-off; we have to be careful to include all local dirs that appear in the path leading up to a `session.Build`.

`WithLocalDirs` is additive and handles deduping to make this pretty easy, but ya still gotta remember to do it.